packages/go/cypher/models/pgsql/format/format.go

```diff
@@ -423,9 +423,17 @@ func formatNode(builder *OutputBuilder, rootExpr pgsql.SyntaxNode) error {
 			exprStack = append(exprStack, *typedNextExpr)
 
 		case pgsql.TypeCast:
-			exprStack = append(exprStack, pgsql.FormattingLiteral(typedNextExpr.CastType), pgsql.FormattingLiteral(")::"))
-			exprStack = append(exprStack, typedNextExpr.Expression)
-			exprStack = append(exprStack, pgsql.FormattingLiteral("("))
+			switch typedNextExpr.Expression.(type) {
+			case pgsql.Parenthetical:
+				// Avoid formatting type-casted parenthetical statements as (('test'))::text - this should instead look like ('test')::text
+				exprStack = append(exprStack, pgsql.FormattingLiteral(typedNextExpr.CastType), pgsql.FormattingLiteral("::"))
+				exprStack = append(exprStack, typedNextExpr.Expression)
+
+			default:
+				exprStack = append(exprStack, pgsql.FormattingLiteral(typedNextExpr.CastType), pgsql.FormattingLiteral(")::"))
+				exprStack = append(exprStack, typedNextExpr.Expression)
+				exprStack = append(exprStack, pgsql.FormattingLiteral("("))
```

Comment:
I realize this is a style choice but why call append multiple times vs just making full use of the power of the variadric functionality?

Reply:
I think this is easier to read, than having one super long line representing the appends

Reply:
I guess you could just break one append into multiple lines, but this seems pretty straightforward too?

Reply:
The `exprStack` will pop elements off but not lose the slice's capacity. The additional appends do incur a cost but the majority of it will be checking the capacity of the slice and doing indexed assignments.

This is just poor ergonomics in the code and should be refactored. I'd like to come back to this in another changeset.

Reply:
Oh, and it's out of order, or feels that way because it's a stack.